### PR TITLE
Fix an issue in parallel execution where inconsistent `FronzenTrial` could be created with low probability.

### PR DESCRIPTION
**Pull request by sile**
This PR resolves the problem reported by https://github.com/pfnet/optuna/pull/349#issuecomment-475086642.

### CODE REVIEW

**Comment by sile** [comment]
@toshihikoyanase Thanks! I updated comments at the commit https://github.com/pfnet/optuna/pull/361/commits/cb5ad795845c443ebde3eeb5514ccf26fa2a0c9c.

File: optuna/storages/rdb/storage.py [hunk 1]
```diff
@@ -514,9 +514,13 @@ def _merge_trials_orm(
...
                 system_attrs[system_attr.key] = json.loads(system_attr.value_json)
 
-            # TODO(Yanase): Use trial.number after TrialModel.number is added.
-            trial_number = self.get_trial_number_from_id(trial_id)
-            result.append(
+            # `0` is a dummy value.
+            # It will be replaced by a proper value before returned to the caller.
+            #
+            # TODO(ohta): Use trial.number after TrialModel.number is added.
+            trial_number = 0
```
Comment (g-votte) [inline optuna/storages/rdb/storage.py]: I think `-1` is safer than `0`, a valid trial number.

Comment (sile) [reply optuna/storages/rdb/storage.py]: That makes sense. Thanks.

Comment (sile) [reply optuna/storages/rdb/storage.py]: Fixed: https://github.com/pfnet/optuna/pull/361/commits/859b2d2e7e2d8d8f727e974164e6d1da4ac5843a

**Pull request merged**